BoutTime/BoutTimeGame.swift

```
@@ -0,0 +1,182 @@
+enum EventError: ErrorType {
+    case InvalidResource
+    case ConversionError
+}
+
+// Helper Classes
+class PlistConverter {
+    class func dictionaryFromFile(resource: String, ofType type: String) throws -> [String: AnyObject] {
+        
+        guard let path = NSBundle.mainBundle().pathForResource(resource, ofType: type) else {
+            throw EventError.InvalidResource
+        }
+        
+        guard let dictionary = NSDictionary(contentsOfFile: path), let castDictionary = dictionary as? [String: AnyObject] else {
+            throw EventError.ConversionError
+        }
+        return castDictionary
+    }
+}
+
+class EventsUnarchiver {
+    class func vendingInventoryFromDictionary(dictionary: [String: AnyObject]) throws -> [EventType] {
+        
+        var events: [EventType] = []
+        
+        for (key, value) in dictionary {
+            var value = value as! [String: AnyObject]
+            let order: Int = value["order"] as! Int
+            let url: String = value["url"] as! String
+            let event = Event(order: order, url: url, name: key)
+            events.append(event)
+        }
+        return events
+    }
+}
+
+
+// Concrete Types
+struct Event: EventType {
+    let order: Int
+    let url: String
+    let name: String
+}
+
+class Round: RoundType {
+    var events: [EventType]
+    var countCorrectEvents = 0
+    
+    required init(events: [EventType]) {
+        self.events = events
+    }
+    
+    func changeOrderEvents(buttonTag: Int) {
+    
+        switch buttonTag {
+        case 0, 1:
+            let firstEvent = events[0]
+            events[0] = events[1]
+            events[1] = firstEvent
+        case 2, 3:
+            let secondEvent = events[1]
+            events[1] = events[2]
+            events[2] = secondEvent
+        case 4, 5:
+            let thirdEvent = events[2]
+            events[2] = events[3]
+            events[3] = thirdEvent
+        default:
+            print("")
```

I was thinking that maybe the `default` should throw an exception? I'm not sure because I don't know exceptions very well.